encodings/fastlanes/src/bitpacking/compress.rs

```diff
@@ -369,9 +377,8 @@ pub unsafe fn unpack_single_primitive<T: NativePType + BitPacking>(
     unsafe { BitPacking::unchecked_unpack_single(bit_width, packed_chunk, index_in_chunk) }
 }
 
-pub fn find_best_bit_width(array: &PrimitiveArray) -> VortexResult<u8> {
-    let bit_width_freq = bit_width_histogram(array)?;
-    best_bit_width(&bit_width_freq, bytes_per_exception(array.ptype()))
+pub fn find_best_bit_width(array: &PrimitiveArray, bit_width_freq: &[usize]) -> VortexResult<u8> {
```

Comment:
you needed the array here before but now you only need a ptype to know the width of the original value. Might be simpler to change the arg?